metrics/amplitude/client.go

```diff
@@ -3,64 +3,46 @@ package amplitude
 import (
 	"bytes"
 	"context"
-	"encoding/json"
+	"fmt"
 	"net/http"
 	"time"
+
+	"github.com/valyala/fastjson"
 )
 
 // Client manages the communication to the Amplitude API
 type Client struct {
 	eventEndpoint string
 	key           string
 	client        *http.Client
+	Arena         *fastjson.Arena
 }
 
-type Event struct {
-	AppVersion      string                 `json:"app_version,omitempty"`
-	DeviceID        string                 `json:"device_id,omitempty"`
-	EventID         int                    `json:"event_id,omitempty"`
-	EventProperties map[string]interface{} `json:"event_properties,omitempty"`
-	EventType       string                 `json:"event_type,omitempty"`
-	Groups          map[string]interface{} `json:"groups,omitempty"`
-	OsName          string                 `json:"os_name,omitempty"`
-	OsVersion       string                 `json:"os_version,omitempty"`
-	Platform        string                 `json:"platform,omitempty"`
-	ProductID       string                 `json:"productId,omitempty"`
-	Quantity        int                    `json:"quantity,omitempty"`
-	SessionID       int64                  `json:"session_id,omitempty"`
-	StartVersion    string                 `json:"start_version,omitempty"`
-	Time            int64                  `json:"time,omitempty"`
-	UserID          string                 `json:"user_id,omitempty"`
-	UserProperties  map[string]interface{} `json:"user_properties,omitempty"`
-}
-
-type EventRequest struct {
-	APIKey string  `json:"api_key,omitempty"`
-	Events []Event `json:"events,omitempty"`
-}
+type Event *fastjson.Value
 
 // New client with API key
 func New(eventEndpoint string, key string) *Client {
+	arenaPool := &fastjson.ArenaPool{}
 	return &Client{
 		eventEndpoint: eventEndpoint,
 		key:           key,
 		client:        new(http.Client),
+		Arena:         arenaPool.Get(),
```

Comment:
I do not put it back because it persists until the application is closed. @cheggaaa

Reply:
then what a purpose?

Reply:
didn't understand how it works at first, simply created an arena